code/extensions/CDNFile.php

```diff
@@ -209,6 +209,10 @@ public function onAfterWrite() {
                 $pathBefore = $changedFields['Filename']['before'];
                 $pathAfter = $changedFields['Filename']['after'];
 
+                if($this->owner instanceof Folder) {
+                    $this->updateChildFolderLinks($this->owner->Children(), $pathBefore, $pathAfter);
```

Comment:
Rather than Children() which is affected by a bunch of stuff in hierarchy, would it be better to use `Folder::get()->filter('ParentID', $this->owner->ID);` ?